util/cron/common-ofi.bash

```diff
@@ -10,23 +10,45 @@ export CHPL_COMM=ofi
 
 # Select a launcher and out-of-band support.  If we're on a Cray XC
 # system this will just fall out automatically.  On a slurm-based system
-# where slurm was configured with the PMIx server-side plugin and we can
-# find a PMIx client-side library, then we can use slurm-srun directly,
-# along with the comm=ofi slurm-pmi2 out-of-band support.  This is seen
-# on Cray CS systems, for example.  Otherwise, our only option is the
-# stopgap mpirun4ofi launcher, in which case we also need MPI and the
-# MPI-based out-of-band support.
+# where slurm was configured with the PMI2 or PMIx server-side plugin
+# and we can find a matching client-side library, then we can use
+# slurm-srun directly, along with the comm=ofi slurm-pmi2 out-of-band
+# support.  This is seen on Cray CS systems, for example.  Otherwise,
+# our only option is the stopgap mpirun4ofi launcher, in which case we
+# also need MPI and the MPI-based out-of-band support.
 lchOK=
 if [[ $($CHPL_HOME/util/chplenv/chpl_platform.py --target) == cray-x* ]] ; then
   lchOK=y
-elif [[ "$($CHPL_HOME/util/chplenv/chpl_launcher.py)" == slurm-srun ]] && \
-     srun --mpi=list 2>&1 | grep -q pmix && \
-     module avail pmix 2>&1 | grep -q pmix ; then
-  module load pmix
-  export CHPL_LAUNCHER=slurm-srun
-  export CHPL_COMM_OFI_OOB=slurm-pmi2
-  export SLURM_MPI_TYPE=pmix
-  lchOK=y
+elif [[ "$($CHPL_HOME/util/chplenv/chpl_launcher.py)" == slurm-srun ]] ; then
+  pmiTypes=$(srun --mpi=list 2>&1)
+  if [[ "$pmiTypes" == *pmi2* ]] ; then
+    # Try to use pmi2 support.
+    srunDir=$(which srun)
+    sLibDir=${srunDir%/bin/srun}/lib64
+    if [[ -d $sLibDir ]] ; then
+      export CHPL_LD_FLAGS="${CHPL_LD_FLAGS:+$CHPL_LD_FLAGS }-L$sLibDir -Wl,-rpath,$sLibDir"
+      export SLURM_MPI_TYPE=pmi2
+      lchOK=y
+    fi
+  fi
+  if [[ "$lchOK" != y && "$pmiTypes" == *pmix* ]] ; then
```

Comment:
Personally, I'd be fine just assuming slurm-pmi2 for now and removing the pmix support since it won't be tested (but leave it in the git history so we can resurrect it if we want/need it later)

Reply:
I'm a bit loathe to do that without running the new stuff for a few weeks anyway.  Also, I have another thing I'd like to make some progress on today, before being away for the rest of the year.  So, can we leave that for the future?  Alternatively, you could grab this fork and do that part; that would be fine with me.

Reply:
Yup, leaving it in for now sounds fine.